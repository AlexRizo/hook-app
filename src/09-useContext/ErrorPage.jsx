import React from 'react'
import { useRouteError } from 'react-router-dom'

export const ErrorPage = () => {
    const error = useRouteError();
    console.log(error);
    
    return (
        <div id='error-page'>
            <h1>Oops!</h1>
            <p>Sorry, an unexpected error has ocurred.</p>
            <p>
                <strong>Error: </strong>
                <i>{ error.statusText || error.message }</i>
            </p>
        </div>
    )
}
