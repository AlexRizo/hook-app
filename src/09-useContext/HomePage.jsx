import React, { useContext } from 'react'
import UserContext from './context/userContext';

export const HomePage = () => {
    const { user } = useContext(UserContext);
    console.log(user);

    return (
        <>
            <h1>HomePage <small>{ user?.name }</small></h1>
            <hr />
            
        </>
    )
}
