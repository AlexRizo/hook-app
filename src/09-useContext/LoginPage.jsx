import { useContext } from 'react'
import UserContext from './context/userContext';

export const LoginPage = () => {

    const { user, setUser } = useContext(UserContext);
    console.log(user);
    
    return (
        <>
            <h1>LoginPage</h1>
            <hr />
            
            <pre>{ JSON.stringify(user, null, 4) }</pre>

            <button 
                className='btn btn-primary'
                onClick={ () => setUser({ id: 123, name: 'David' }) }    
            >
                Establecer usuario
            </button>
        </>
    )
}
