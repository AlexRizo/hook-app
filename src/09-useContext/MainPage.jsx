import React from 'react'
import { Outlet } from 'react-router-dom'
import { Navbar } from './Navbar'
import { UserProvider } from './context/UserProvider'

export const MainPage = () => {
    return (
        <UserProvider>
            <h1>MainPage</h1>
                <Navbar />
            <hr />

            <Outlet/>
        </UserProvider>
    )
}
