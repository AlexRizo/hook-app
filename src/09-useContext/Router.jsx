import { createBrowserRouter } from 'react-router-dom';
import { MainPage, AboutPage, HomePage, LoginPage, ErrorPage } from './';

const Router = () => {
    const router = createBrowserRouter([
        { 
            path: '/',
            element: <MainPage/>,
            errorElement: <ErrorPage/>,
            children: [
                { path: 'home', element: <HomePage/> },
                { path: 'about', element: <AboutPage/> },
                { path: 'login', element: <LoginPage/> },
            ]
        },
    ]);

    return router;
}

export default Router;