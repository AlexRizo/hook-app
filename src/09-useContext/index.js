export * from './MainPage';
export * from './HomePage';
export * from './AboutPage';
export * from './LoginPage';

export * from './ErrorPage';