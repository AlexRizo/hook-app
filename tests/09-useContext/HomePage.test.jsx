import { render, screen } from "@testing-library/react";
import { HomePage } from "../../src/09-useContext/HomePage";
import UserContext from "../../src/09-useContext/context/userContext";

describe('Pruebas en <HomePage/>', () => {
    const user = {
        id: 1,
        name: 'David'
    };
    
    test('debe de mostrar el componente correctamente (sin el usuario)', () => {
        render(
            <UserContext.Provider value={ { user: null } }>
                <HomePage/>
            </UserContext.Provider>
        );

        const preTag = screen.getByLabelText('pre'); // aria-label
        expect(preTag.innerHTML).toBe('null');
    });

    test('debe de mostrar el componente correctamente (con el usuario)', () => {
        render(
            <UserContext.Provider value={ { user } }>
                <HomePage/>
            </UserContext.Provider>
        );

        const preTag = screen.getByLabelText('pre'); // aria-label
        expect(preTag.innerHTML).toBe(JSON.stringify(user, null, 4));

    });
})