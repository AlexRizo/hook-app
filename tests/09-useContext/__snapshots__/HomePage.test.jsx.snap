// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <HomePage/> debe de mostrar el componente correctamente (sin el usuario) 1`] = `
[
  <pre
    aria-label="pre"
  >
    null
  </pre>,
]
`;
